### PR TITLE
Updated documentation and added conversion utility

### LINKED ISSUE #34221: Documentation improvement on tiktoken integration
I want to load tokenizers from tiktoken to use for model training. Right now, this page https://huggingface.co/docs/transformers/main/en/tiktoken is only info I could find on tiktoken integration.

Following the instructions on the page, I just need to add a `tokenizer.model` to model directory. In light of this, I have a few questions:
1. How do I generate this file from tiktoken's Encoding class?
2. What happens when I save the model? Does it get converted to HF format or is it saved as tiktoken tokenizer?

Would appreciate any help on this!

### PR DESCRIPTION
**Pull request by viktor-shcherb**
# What does this PR do?

Documentation improvement on tiktoken integration + tiktoken conversion function.

Fixes #34221 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests? 

## Who can review?

@ArthurZucker
@stevhliu

### CODE REVIEW

File: docs/source/en/tiktoken.md
Comment (stevhliu) [inline docs/source/en/tiktoken.md:44]: Maybe we can create a new section here that describes the purpose and motivation a little more.

```suggestion
## Create tiktoken tokenizer

The `tokenizer.model` file contains no information about additional tokens or pattern strings. If these are important, convert the tokenizer to `tokenizer.json`, the appropriate format for [`PreTrainedTokenizerFast`].

Generate the `tokenizer.model` file with [tiktoken.get_encoding](https://github.com/openai/tiktoken/blob/63527649963def8c759b0f91f2eb69a40934e468/tiktoken/registry.py#L63) and then convert it to `tokenizer.json` with [`convert_tiktoken_to_fast`].
```

File: docs/source/en/tiktoken.md
Comment (stevhliu) [inline docs/source/en/tiktoken.md:58]:
```suggestion
The resulting `tokenizer.json` file is saved to the specified directory and loaded with [`PreTrainedTokenizerFast`].

```py
PreTrainedTokenizerFast.from_pretrained("config/save/dir")
```
```

**Review by stevhliu — APPROVED** [review]
Thanks, docs LGTM! Once @ArthurZucker has reviewed, we can merge 🙂

File: src/transformers/tokenization_utils_fast.py
Comment (ArthurZucker) [inline src/transformers/tokenization_utils_fast.py:899]: very nice! 
We should do :
```python
from tiktoken import get_encoding

# You can load your custom encoding or the one provided by OpenAI
encoding = get_encoding("gpt2")
```
in this function directly IMO! 

Also let's maybe place this under the `integration` folder ! 🤗

**Review by ArthurZucker — COMMENTED** [review]
NIce PR!

File: src/transformers/tokenization_utils_fast.py
Comment (ViktorooReps) [reply src/transformers/tokenization_utils_fast.py:899]: I wouldn't like to load the encoding inside the function to allow for custom encodings like https://github.com/openai/tiktoken?tab=readme-ov-file#extending-tiktoken 

We could allow `Encoding | str` as input and load the encoding with `get_encoding` if a str is passed.

Comment (ViktorooReps) [reply src/transformers/tokenization_utils_fast.py]: Moved it to `integration/tiktoken.py`

File: src/transformers/integrations/tiktoken.py
Comment (ArthurZucker) [inline src/transformers/integrations/tiktoken.py:44]: let's split in two lines: 
```suggestion
    tokenizer = TikTokenConverter(
        vocab_file=save_file_absolute, pattern=encoding._pat_str, additional_special_tokens=encoding._special_tokens
    ).tokenizer()
    tokenizer.save(output_file_absolute)
```

File: src/transformers/tokenization_utils_fast.py
Comment (ArthurZucker) [inline src/transformers/tokenization_utils_fast.py:897]: let's revert this as it's not needed!

**Review by ArthurZucker — COMMENTED** [review]
LGTM was not pinged so did not come back to it! Could you run `make fixup` and rebase to make sure you are up to date?

**Comment by ViktorooReps** [comment]
@ArthurZucker should be all good now

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

File: src/transformers/integrations/tiktoken.py
Comment (EndlessReform) [inline src/transformers/integrations/tiktoken.py:44]: To actually work, the tokenizer pipeline needs the ByteLevel and Split pre-tokenizers (with the pattern string), the ByteLevel post-processor, and the special tokens from the original encoding. It looks like the `.converted()` method on the original TikTokenConverter adds these already: https://github.com/huggingface/transformers/blob/98e8062df3cf82b367e8a243963e4afb0d9d3407/src/transformers/convert_slow_tokenizer.py#L1532

but the `convert_tiktoken_to_fast` wrapper just uses the `.tokenizer()` method, which saves only the actual merges. This isn't documented, so unless you read the source code and/or inspect the actual tokenizers file, you won't realize the output is incomplete until Unicode tokens or special tokens start breaking.

This took me a couple hours to figure out what went wrong (porting the Fish Speech 1.5 tokenizer from Tiktoken). Is there a reason `convert_tiktoken_to_fast` doesn't just use the `.converted()` method? If this isn't possible, could the docs at least explicitly call out what the remaining steps are to set up the BPE pipeline?

Comment (ArthurZucker) [reply src/transformers/integrations/tiktoken.py:44]: Sorry about that, you are right! pushing a fix

**Pull request merged**
